### PR TITLE
#277 - Migrate to .Net Core 3.1

### LINKED ISSUE #277: Migrate to .net core 3.1
Migrating to .net core 3.1 gives us some new possibilities like:

 * Using new interfaces (e.g. IAsyncDisposable, solves https://github.com/camunda-community-hub/zeebe-client-csharp/issues/39)
 * Using new grpc lib (dotnet-grpc) https://grpc.io/blog/grpc-on-dotnetcore/ , which would solve https://github.com/camunda-community-hub/zeebe-client-csharp/issues/194 and gives us better performance https://github.com/LesnyRumcajs/grpc_bench/wiki/2021-04-13-bench-results
 * tbd

Furthermore .net core 2.1 is as LTS only supported until August 21, 2021, see https://dotnet.microsoft.com/platform/support/policy/dotnet-core

### PR DESCRIPTION
**Pull request by Christian-Oleson**
closes #277

### CODE REVIEW

File: Client.Examples/Client.Examples.csproj [hunk 1]
```diff
@@ -13,7 +13,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="3.1.19" />
     <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="5.0.2" />
-    <PackageReference Include="Microsoft.Extensions.Logging" Version="5.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Logging" Version="3.1.19" />
```
Comment (Zelldon) [inline Client.Examples/Client.Examples.csproj:16]: Hm why this was dropped to a lower version ? :thinking:

**Review by Zelldon — COMMENTED** [review]
Thanks @Christian-Oleson 

could you please adjust your commit messages and follow the contribution guide https://github.com/camunda-cloud/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines

File: Client.Examples/Client.Examples.csproj (hunk 1 above)
Comment (Christian-Oleson) [reply Client.Examples/Client.Examples.csproj:16]: There were two completely different versions of Microsoft.Extensions.Logging here. I also found a lower version in Client.IntegrationTests.csproj. We should avoid multiple versions, so this consolidates Client.IntegrationTests.csproj from 3.1.14 to 3.1.19 while consolidating the 5.x version to 3.1.19. I would suggest moving both projects to 5.x when you move to 5.x (or really .NET 6)

Comment (Christian-Oleson) [reply Client.Examples/Client.Examples.csproj:16]: > Thanks @Christian-Oleson
> 
> could you please adjust your commit messages and follow the contribution guide https://github.com/camunda-cloud/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines

I think I fixed the commit message? Let me know if I'm misunderstanding anything.

**Comment by Zelldon** [comment]
Thanks @Christian-Oleson still the title of the commit (or subject however you want to call it) is wrong, but should be fine for now. just for the next time :) 

`feat: Migrate to .Net Core 3.1`  is the subject

**Pull request merged**